src/app/shared/util/metrics.util.ts

```diff
@@ -22,9 +24,34 @@ export const filterEpisodes = (filter: FilterModel, episodes: EpisodeModel[]) =>
   }
 };
 
-export const filterPodcastMetrics = (filter: FilterModel, podcastMetrics: PodcastMetricsModel[]): PodcastMetricsModel => {
-  if (filter.podcast && filter.interval && podcastMetrics) {
-    const metrics = podcastMetrics.filter(metric => metric.seriesId === filter.podcast.seriesId);
+export const filterMetricsByDate = (beginDate: Date, endDate: Date, interval: IntervalModel, metrics: any[][]): any[][] => {
+  const findEntryByDate = (date: Date) => {
+    return metrics.findIndex(m => {
+      return new Date(m[0]).valueOf() === roundDateToInterval(date, interval).valueOf();
+    });
+  };
+  const begin = findEntryByDate(beginDate);
+  const end = findEntryByDate(endDate);
+  if (begin !== -1 && end !== -1) {
+    return metrics.slice(begin, end + 1);
+  } else {
+    return null; // no partial data
+  }
+};
+
+export const findPodcastMetrics = /* TODO: findPodcastMetrics? */
```

Comment:
this line seems like it doubts itself

Reply:
oh, I did end up renaming that function and didnt remove the comment about it